File: 226/invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if None == root:
            return None
        if None != root.left:
            self.invertTree(root.left)
        if None != root.right:
            self.invertTree(root.right)
        root.left, root.right = root.right, root.left
        return root

File: 226/InvertBinaryTree.scala
object InvertBinaryTree {
  def invertTree(root: TreeNode): TreeNode = {
    if (root != null) {
      val left = invertTree(root.left)
      val right = invertTree(root.right)
      root.left = right
      root.right = left
    }
    root
  }

  class TreeNode(var _value: Int = 0) {
    var value: Int = _value
    var left: TreeNode = null
    var right: TreeNode = null
  }

}
